SCClassLibrary/JITLib/Patterns/Pdef.sc

```diff
@@ -274,11 +274,14 @@ Pdefn : PatternProxy {
 	}
 
 	storeArgs { ^[key] } // assume it was created globally
+
 	copy { |toKey|
-		if(key == toKey) { Error("cannot copy to identical key").throw };
+		if(toKey.isNil or: { key == toKey }) { Error("cannot copy to identical key").throw };
 		^this.class.new(toKey).copyState(this)
 	}
 
+	dup { |n = 2| ^{ this }.dup(n) }
```

Comment:
`dup` appears to be a new feature, not used in the `copy` fix. Probably cleaner to remove `dup` from this PR and submit a feature request separately. (Or, if that's inconvenient, then document and unit-test the new method.)

Reply:
I see your point. On the other hand, this is a bugfix PR ("class library: fix def proxy dup") which addresses the current problem. I had a version where the proxy would just return itself for copy when the key is nil, which is kind of nice, but then I thought that this may cause trouble in other use cases where you really expect a proper copy. What do you think?

I don't think we need to write a test, because the method reads more clearly than any test I could think of.

Reply:
Ok, I see, it's not "new" in the sense that these classes answered to `dup` before, just with an inherited implementation. (But `dup` here isn't copying, is it? `{}.dup` is a synonym for Array.fill, which evaluates the function fresh every time, giving you distinct instances of `.new` objects but it doesn't copy references to existing objects. How are they related?)

I agree that it's better not to return a non-copy for a nil key.

Reply:
When you write `Pdef(\x).dup(3)` you expect to get `[Pdef(\x), Pdef(\x), Pdef(\x)]`, and because of the nature of these "def" proxies, they are all exactly the same object.

An alternative for dup would be:
```
{ |n=2| { ^this.class.new(this.key) }.dup(n) }
```
But that just threads through the new method and returns the same thing.

Reply:
Ok, I see now -- I thought Object:dup didn't copy, but it does. I wondered what was the purpose of redirecting Pdef etc `dup` to Function:dup -- might be nice to explain in a comment that this is to avoid the `copy` in Object:dup.